cfme/physical/provider/__init__.py

```diff
@@ -28,14 +37,30 @@ def __init__(
         self.name = name
         self.key = key
 
-    @variable(alias='db')
+    @variable(alias='ui')
     def num_server(self):
-        pass
+        try:
+            num = self.get_detail('Relationships', 'Physical Servers')
+        except sel.NoSuchElementException:
+            logger.error("Couldn't find number of servers")
+        return int(num)
 
-    @num_server.variant('ui')
-    def num_server_ui(self):
-        pass
+    def delete(self, cancel=True):
+        """
+        Deletes a provider from CFME
 
+        Args:
+            cancel: Whether to cancel the deletion, defaults to True
+        """
+        view = navigate_to(self, 'Details')
+        item_title = version.pick({'5.9': 'Remove this {} Provider from Inventory',
+                                   version.LOWEST: 'Remove this {} Provider'})
+        view.toolbar.configuration.item_select(item_title.format("Infrastructure"),
+                                               handle_alert=not cancel)
+        if not cancel:
+            msg = ('Delete initiated for 1 {} Provider from '
```

Comment:
This is a fairly recent decision for the framework, no fault of yours here.

Can you please move the flash message assertion into the CRUD test, and only call `view.flash.assert_no_error()` here in the CRUD method.  This allows us to still assert that there wasn't any generated error during the CRUD action, but won't block further testing when the flash message isn't being asserted, or shouldn't actually block a test.